### PR TITLE
feature/158_create_cosmos_tidoop_api

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #158
* (unofficial) e2e tests passed:
```
$ curl -X GET "http://195.235.93.173:12000/tidoop/v1/version" | python -m json.tool
{
    "version": "0.1.1"
}
$ curl -X POST "http://195.235.93.173:12000/tidoop/v1/user/frb/test_mr_job" -d '{"jar_path":"testpath", "class_name": "dummyClass"}' -H 'Content-Type: application/json' | python -m json.tool
{
    "job_id": "tidoop_job_1456314516663"
}
$ curl -X GET "http://195.235.93.173:12000/tidoop/v1/user/frb/jobs/tidoop_job_1456314516663" | python -m json.tool
{
    "end_time": "0000-00-00 00:00:00",
    "job_id": "tidoop_job_1456314516663",
    "job_type": "undefined",
    "map_progress": 0,
    "reduce_progress": 0,
    "start_time": "Wed Feb 24 2016 13:48:45 GMT+0100 (CET)"
}
```
* Assignee @pcoello25

### CODE REVIEW

File: cosmos-tidoop-api/README.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+[Top](#top)
+
+##<a name="section2"></a>Installation
+This is a software written in JavaScript, specifically suited for [Node.js](https://nodejs.org) (<i>JavaScript on the server side</i>). JavaScript is an interpreted programming language thus it is not necessary to compile it nor build any package; having the source code downloaded somewhere in your machine is enough.
+
+###<a name="section2.1"></a>Prerequisites
+This REST API has no sense if tidoop-mr-lib is not installed. And tidoop-mr-lib has only sense in a [Hadoop](http://hadoop.apache.org/) cluster, thus both the library and Hadoop are required.
+
+As said, cosmos-tidoop-api is a Node.js application, therefore install it from the official [download](https://nodejs.org/download/). An advanced alternative is to install [Node Version Manager](https://github.com/creationix/nvm) (nvm) by creationix/Tim Caswell, whcih will allow you to have several versions of Node.js and switch among them.
```
Comment (pcoello25) [inline cosmos-tidoop-api/README.md]: whcik -> which

File: cosmos-tidoop-api/README.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    {"version": "0.1.1"}
+    
+cosmos-tidoop-api typically listens in the TCP/12000 port, but you can change if by editing `conf/cosmos-tidoop-api.conf` as seen above.
+
+[Top](#top)
+
+##<a name="section5"></a>Usage
+Please refer to this [Apiary](http://docs.tidoopmrlibapi.apiary.io/#) documentation.
+
+Please observe the MapReduce jobs usually take some time to return a result. This is why the cosmos-tidoop-api operations run the job, but do not return any result, except a `200 OK` (if the job could be successfully run) and a job identifier. In order to get the result of the opearion (or its progress), such a job identifier must be used for querying the API later.
```
Comment (pcoello25) [inline cosmos-tidoop-api/README.md]: opearion -> operation

File: cosmos-tidoop-api/README.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+* ***The new job could not be added to the database***: This message may appear once a MapReduce job has been launched and it is wanted to be tracked throught the database. Most probably the connection with database has been lost, or the MySQL service is unavailable.
+* ***The MR job could not be run***: This message may appear when a new MapReduce job from tidoop-mr-lib is wanted to be run. Please check the logs generated by the library since it is not a problem related to the API.
+* ***Could not get job information for the given job_id***: This message may appear when the status/progress of a job is requested. Most probably the connection with database has been lost, or the MySQL service is unavailable.
+
+[Top](#top)
+
+###<a name="section6.2"></a>Database
+Information regarding launched jobs can be found in a MySQL table named `tidoop_job` within a database named `cosmos` in the MySQL deployment you did when installing cosmos-tidoop-api. Such a table contains the jod ID, the job type, the stating and ending timestamp the progress in terms of map and reduce percentage.
```
Comment (pcoello25) [inline cosmos-tidoop-api/README.md]: stating -> starting

File: cosmos-tidoop-api/README.md (hunk 1 above)
Comment (frbattid) [inline cosmos-tidoop-api/README.md]: Fixed in https://github.com/telefonicaid/fiware-cosmos/commit/a46e1731daedd1c5d9c75e1ca08e90be37aff51a

File: cosmos-tidoop-api/README.md (hunk 2 above)
Comment (frbattid) [inline cosmos-tidoop-api/README.md]: Fixed in https://github.com/telefonicaid/fiware-cosmos/commit/a46e1731daedd1c5d9c75e1ca08e90be37aff51a

File: cosmos-tidoop-api/README.md (hunk 3 above)
Comment (frbattid) [inline cosmos-tidoop-api/README.md]: Fixed in https://github.com/telefonicaid/fiware-cosmos/commit/a46e1731daedd1c5d9c75e1ca08e90be37aff51a

**Comment by pcoello25** [comment]
LGTM

**Pull request merged**